department-of-veterans-affairs/caseflow
# Sort by veteran name

Resolves #11312

### Description
Allows the back end to sort by the veteran's name (last then first). Also adds veteran name to the cached appeal attributes table.


|**Before** | **ASC** | **DESC**
|--- | --- | --- | 
|<img width="361" alt="Screen Shot 2019-08-06 at 2 47 25 PM" src="https://user-images.githubusercontent.com/45575454/62568136-93568880-b85a-11e9-9b7d-e10f0f8a0872.png">|<img width="363" alt="Screen Shot 2019-08-06 at 2 48 52 PM" src="https://user-images.githubusercontent.com/45575454/62568117-8b96e400-b85a-11e9-833d-577071d0651f.png">|<img width="363" alt="Screen Shot 2019-08-06 at 2 48 44 PM" src="https://user-images.githubusercontent.com/45575454/62568127-905b9800-b85a-11e9-9c35-6b5b227d7f58.png">

### Acceptance Criteria

- [ ]   Back end correctly stores each appeal's veteran's name
- [ ]   Front end correctly shows the back end sorted tasks

### Testing plan

1. Run the new migration with
```bash
bin/rails db:migrate
```
2. Run the updated job in the rails console with
```ruby
> UpdateCachedAppealsAttributesJob.perform_now
> CachedAppeal.find_by(appeal_id: 11, appeal_type: Appeal.name)
=> => #<CachedAppeal:0x00007f8668e123c8
 appeal_id: 11,
 appeal_type: "Appeal",
 closest_regional_office_city: "St. Petersburg",
 docket_number: "190805-11",
 docket_type: "direct_review",
 vacols_id: nil,
 veteran_name: "Goodwin, Leo">
```
3. Change user to BVALSPORER and go to the /organizations/vlj-support "On hold" tab
4. Toggle the case details sorting button (Note that the names should sort on "#{last_name}, #{first_name}")

### Schema Changes
